vscode-lean4/src/leanclient.ts

```diff
@@ -163,9 +160,6 @@ export class LeanClient implements Disposable {
                 }
             },
         }
-        this.setProgress(Object.assign({}, ...workspace.textDocuments
```

Comment:
Removing this changes the semantics slightly in that yellow bars only appear when the server actually begins compiling the header. I think this is nice, as visually indicates things are working.

Reply:
I don't have extremely strong feelings about this, but it was very much intentional.  When opening a Lean file, I'm interested in whether there are errors or not.  The progress bars allow me to distinguish the states of "no errors because Lean hasn't compiled this part yet" and "no errors because the proof is fine".

This is the same reason why I said that the fileProgress notification should be sent *before* processing is started (in https://github.com/leanprover/lean4/issues/503).  The important information is not that Lean is still running, but that it is not finished yet.

Reply:
I see your point. The only real difference is that with this change, if the server fails to start the gutters won't appear, whereas before they'd be there even though nothing is actually being processed. You still know that the lack of errors is due to server failure because there will be the "server failed to start" popup. IMO this way it's clearer when things have failed, and when things are actually compiling (where we do send server notifs before starting compilation as you note), and you can still clearly tell why there are no errors. But if you still disagree, I'm happy to revert this change.

Reply:
Ah, if this only happens with a dead server then it's a non-issue.  In that case you already get a lot of errors anyhow.

Reply:
Well, there are also no gutters between `exec lean --server` and "`lean --worker` sends `progress @ 0`". This is a few ms right now, but I suppose it could be a problem with a bad plugin which takes long to initialize. We could show [progress](https://github.com/microsoft/vscode-extension-samples/tree/main/progress-sample/src) then.